templates/serverpod_templates/github/workflows/deployment-docker.md

```diff
@@ -0,0 +1,344 @@
+
+For security reasons, we will create a new user to manage the deployment. This
+user will not have root privileges.
+
+#### Step 1: Create the new user
+
+```bash
+sudo adduser github-actions
+```
+
+Replace `github-actions` with your desired username. This command will prompt
+you to set a password and fill in user information.
+
+#### Step 2: Grant Docker permissions
+
+Add the user to the `docker` group, so they can run Docker commands:
+
+```bash
+sudo usermod -aG docker github-actions
+```
+
+#### Step 3: Enable SSH access
+
+The SSH access should be available by default for any user on the server.
+However, to ensure they can access it, check the `sshd_config` file:
+
+```bash
+sudo nano /etc/ssh/sshd_config
+```
+
+Find or add the `AllowUsers` directive in the file. This directive specifies which
+users are allowed to SSH into the server. If it doesn't exist, add it at the end
+of the file. If there are multiple users, separate them with spaces:
+
+```text
+AllowUsers root github-actions
+```
+
+To save and exit the file, press `Ctrl + X`, then `Y`, and finally `Enter`.
+Save the file and restart the SSH service to apply changes:
+
+```bash
+sudo systemctl restart ssh
+```
+
+#### Step 4: Set up SSH key-based authentication
+
+1. Log in as the new user:
+
+   ```bash
+   su - github-actions
+   ```
+
+2. Create a ssh keypair:
+
+   ```bash
+   ssh-keygen -t rsa -b 4096
+   ```
+
+   Leave any options at their default values by pressing enter.
+
+3. Add the public SSH key to the `authorized_keys` file:
+
+   ```bash
+   cat ~/.ssh/id_rsa.pub >> ~/.ssh/authorized_keys
+   ```
+
+4. Restart the SSH service to apply changes:
+
+   ```bash
+   sudo systemctl restart ssh
+   ```
+
+Copy the private key to the clipboard, including the lines `-----BEGIN OPENSSH PRIVATE KEY-----` and ending with `-----END OPEN SSH PRIVATE KEY-----`. Save this key in a secure place, you will need it later. Use the following command to display the private key:
+
+```bash
+cat ~/.ssh/id_rsa
+```
+
+### Firewall configuration
+
+In the Hetzner Web Interface, enter your server configuration and click on "Firewalls", then click on "Create Firewall".
+
+By default there will be two inbound rules, one for SSH (Which has Port 22) and one for ICMP (which has protocol set to ICMP).
+
+We will add two more for HTTP and HTTPS.
+
+1. Click on "Add Rule", name it HTTP and set the port to 80 and the protocol to TCP.
+2. Click on "Add Rule", name it HTTPS and set the port to 443 and the protocol to TCP.
+3. Make sure in "apply to" section your server is selected.
+4. Click on "Create Firewall".
+
+## Preparing the domain
+
+In order to be able to access your server, you need to have a domain.
+You can buy a domain from any domain provider, e.g., [Namecheap](https://www.namecheap.com/) or [GoDaddy](https://www.godaddy.com/).
```

Comment:
Comment: This is more of a personal opinion from me but I try to avoid using or linking to GoDaddy after their actions around sopa/pipa (https://en.wikipedia.org/wiki/Protests_against_SOPA_and_PIPA), perhaps this is starting to become ancient history.